File: lib/data/annonce_objet_categories.dart
/// Catégories des petites annonces « objets & matériel » liées aux animaux
/// (profil particulier). Aucune catégorie ne concerne un animal vivant.
///
/// Doit rester aligné avec `website/src/lib/annonce-objet-categories.ts`.
class AnnonceObjetCategorie {
  final String slug;
  final String label;
  final String emoji;
  final String exemples;
  const AnnonceObjetCategorie(this.slug, this.label, this.emoji, this.exemples);
}

const kAnnonceObjetCategories = <AnnonceObjetCategorie>[
  AnnonceObjetCategorie('habitat', 'Habitat & cage', '🏠',
      'Cage, clapier, volière, aquarium, terrarium, niche, box, abri'),
  AnnonceObjetCategorie('accessoires', 'Accessoires', '🦮',
      'Harnais, laisse, collier, gamelle, sac de transport, jouets, couchage'),
  AnnonceObjetCategorie('alimentation', 'Alimentation & fourrage', '🌾',
      'Foin, paille, granulés, litière, compléments'),
  AnnonceObjetCategorie('entretien', 'Entretien & soin', '✂️',
      'Matériel de toilettage, tondeuse, pharmacie, brosses'),
  AnnonceObjetCategorie('terrain', 'Terrain & pâture', '🌳',
      'Location de prairie, parcelle, pré, box en écurie, stabulation, pension'),
  AnnonceObjetCategorie('materiel_agricole', 'Matériel agricole', '🚜',
      "Tracteur, remorque, clôture, abreuvoir, matériel d'élevage"),
  AnnonceObjetCategorie('autre', 'Autre (lié aux animaux)', '📦',
      'Tout autre objet ou service lié aux animaux (pas un animal)'),
];

const kAnnonceObjetTransactions = <String, String>{
  'vente': 'Vente',
  'location': 'Location',
  'don': 'Don',
  'recherche': 'Recherche',
};

const kAnnonceObjetEtats = <String, String>{
  'neuf': 'Neuf',
  'tres_bon': 'Très bon état',
  'bon': 'Bon état',
  'usage': 'État d\'usage',
};

String annonceObjetCategorieLabel(String? slug) {
  for (final c in kAnnonceObjetCategories) {
    if (c.slug == slug) return c.label;
  }
  return 'Autre';
}

String annonceObjetCategorieEmoji(String? slug) {
  for (final c in kAnnonceObjetCategories) {
    if (c.slug == slug) return c.emoji;
  }
  return '📦';
}

/// Libellé de prix affichable pour une ligne `annonces_objets`.
String annonceObjetPrixLabel(Map<String, dynamic> r) {
  final t = (r['type_transaction'] ?? 'vente').toString();
  if (t == 'don') return 'Don';
  if (t == 'recherche') return 'Recherche';
  final p = r['prix'];
  if (p == null) return 'Prix à convenir';
  final unite = (r['prix_unite'] ?? '').toString();
  final neg = r['prix_negociable'] == true ? ' (négociable)' : '';
  return '${(p as num).toStringAsFixed(0)} €$unite$neg';
}

/// Le boost d'une annonce objet est-il encore actif ?
bool annonceObjetBoostActif(dynamic boostUntil) {
  final s = boostUntil?.toString();
  if (s == null || s.isEmpty) return false;
  final d = DateTime.tryParse(s);
  return d != null && d.isAfter(DateTime.now());
}

File: website/src/lib/annonce-objet-categories.ts
// Catégories des petites annonces « objets & matériel » liées aux animaux
// (profil particulier). Aucune catégorie ne concerne un animal vivant.
// Doit rester aligné avec `lib/data/annonce_objet_categories.dart`.

export interface AnnonceObjetCategorie {
  slug: string;
  label: string;
  emoji: string;
  exemples: string;
}

export const ANNONCE_OBJET_CATEGORIES: AnnonceObjetCategorie[] = [
  { slug: 'habitat', label: 'Habitat & cage', emoji: '🏠',
    exemples: 'Cage, clapier, volière, aquarium, terrarium, niche, box, abri' },
  { slug: 'accessoires', label: 'Accessoires', emoji: '🦮',
    exemples: 'Harnais, laisse, collier, gamelle, sac de transport, jouets, couchage' },
  { slug: 'alimentation', label: 'Alimentation & fourrage', emoji: '🌾',
    exemples: 'Foin, paille, granulés, litière, compléments' },
  { slug: 'entretien', label: 'Entretien & soin', emoji: '✂️',
    exemples: 'Matériel de toilettage, tondeuse, pharmacie, brosses' },
  { slug: 'terrain', label: 'Terrain & pâture', emoji: '🌳',
    exemples: 'Location de prairie, parcelle, pré, box en écurie, stabulation, pension' },
  { slug: 'materiel_agricole', label: 'Matériel agricole', emoji: '🚜',
    exemples: "Tracteur, remorque, clôture, abreuvoir, matériel d'élevage" },
  { slug: 'autre', label: 'Autre (lié aux animaux)', emoji: '📦',
    exemples: 'Tout autre objet ou service lié aux animaux (pas un animal)' },
];

export const ANNONCE_OBJET_TRANSACTIONS: Record<string, string> = {
  vente: 'Vente',
  location: 'Location',
  don: 'Don',
  recherche: 'Recherche',
};

export const ANNONCE_OBJET_ETATS: Record<string, string> = {
  neuf: 'Neuf',
  tres_bon: 'Très bon état',
  bon: 'Bon état',
  usage: "État d'usage",
};

export function categorieLabel(slug?: string | null): string {
  return ANNONCE_OBJET_CATEGORIES.find(c => c.slug === slug)?.label ?? 'Autre';
}

export function categorieEmoji(slug?: string | null): string {
  return ANNONCE_OBJET_CATEGORIES.find(c => c.slug === slug)?.emoji ?? '📦';
}
